aea/protocols/dialogue/base.py

```diff
@@ -611,6 +611,51 @@ def reply(
 
         return reply
 
+    def reply_with_target(
```

Comment:
An overload of reply which instead of specifying the target message, you just specify the target message_id.

Reply:
DRY! don't repeat all the code please...

Also, on review, why not simply add another optional argument to `reply`? This is not a neat implementation at all at present.

Reply:
Done